drivers/hmis/app/graphql/types/hmis_schema/ce_referral_source_enrollment.rb

```diff
@@ -129,9 +129,9 @@ def assessments
         joins(:definition).where(definition: { identifier: object.definition_identifiers })
 
       # We don't expect there to be a lot, so calculate in-memory the latest assessment per definition
-      assessments.group_by(&:definition).map do |_, group|
+      assessments.group_by { |assessment| assessment.definition.identifier }.map do |_, group|
         group.max_by { |assessment| [assessment.assessment_date, assessment.id] }
-      end
+      end.sort { |a, b| b.assessment_date <=> a.assessment_date } # sort by assessment date, descending
```

Comment:
Interesting I have never seen this syntax before. It does look like there are handful of places we use it in the codebase already, so its not a totally new pattern, but new to me! The array should be very small (just a handful of items max) so I would expect `sort_by(&:assessment_date).reverse` to perform just fine and is a bit more readable, but this seems OK too.